nettest/src/main.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//======================================================================================================================
+// Imports
+//======================================================================================================================
+
+use anyhow::Result;
+use clap::{Arg, ArgMatches, Command};
+use std::{
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+//======================================================================================================================
+// Standalone Functions
+//======================================================================================================================
+
+fn main() -> Result<()> {
+    // Merge all lines
+    let matches: ArgMatches = Command::new("app")
+        .author("Pedro Henrique Penna <ppenna@microsoft.com>")
+        .about("Network Testing Tool for Demikernel")
+        .arg(
+            Arg::new("src")
+                .long("src")
+                .value_parser(clap::value_parser!(String))
+                .required(true)
+                .value_name("filename")
+                .help("Source file"),
+        )
+        .arg(
+            Arg::new("verbose")
+                .long("verbose")
+                .short('v')
+                .value_parser(clap::value_parser!(bool))
+                .required(true)
+                .help("Verbose output?"),
+        )
+        .get_matches();
+
+    let verbose: bool = *matches.get_one::<bool>("verbose").expect("missing verbose flag");
+    let filename: &String = matches.get_one::<String>("src").expect("missing filename");
+
+    let lines: Vec<String> = read_input_file(filename)?;
+
+    // Process all lines of the source file.
+    for line in lines {
+        if verbose {
+            println!("Line: {:?}", line);
+        }
+        // Process in multiple passes.
+        for pass in ["Lexical Analysis", "Syntax Analysis"] {
```

Comment:
Do you anticipate having more passes apart from the Lexical and Syntax analysis?

Reply:
It may be possible. For instance it would be nice to add a pre-processing step to substitute compile-time constants. Why the question?

Reply:
`Preprocessing` layer sounds good. I was curious about why we need the loop on Ln. 53. But this answers my question.